iib/web/healthcheck.py

```diff
@@ -0,0 +1,202 @@
+        except kombu.exceptions.OperationalError:
+            return False, "Broker connection failed"
+        except kombu.exceptions.KombuError:
+            return False, "An error occurred"
+
+    return True, None
+
+
+def _ping_celery_workers(retries):
+    """
+    Attempt to ping worker, retry on ConnectionError.
+
+    :param int retries: number of retry/retries before returning an empty response
+    :return: dict of {hostname: reply} as returned by ping()
+    """
+    for i in range(retries + 1):
+        if i > 0:
+            time.sleep(0.25 * 2 ** (i - 1))
+        try:
+            replies = app.control.inspect().ping()
+        except ConnectionError:
+            continue
+
+        return replies or {}
+
+    return {}
+
+
+def workers_status(retries=2):
+    """
+    Ping workers, check received replies.
+
+    :param int retries: number of retry/retries on ConnectionError when pinging workers
+    :return: list of workers status information (empty if no replies)
+    """
+    replies = _ping_celery_workers(retries=retries)
+
+    # replies is a dict of {hostname: reply}, convert to a sorted list of [(hostname, reply)]
+    reply_tuples = sorted(replies.items(), key=lambda kv: kv[0])
+    workers = []
+
+    for worker_name, reply in reply_tuples:
+        # the reply is in the following format {'ok': 'pong'}.
+        worker = {"name": worker_name, "ok": "ok" in reply}
+
+        if "ok" not in reply:
+            worker["reason"] = reply.get("error", "unknown reason")
+
+        workers.append(worker)
+
+    return workers
+
+
+def healthcheck(*, short=False, worker_ping_retries=2):
+    """
+    Get status of IIB workers and services that IIB depends on.
+
+    Display the current version of podman, skopeo, opm.
+    :param bool short: raise an error as soon as any problem is found
+    :param int worker_ping_retries: number of retry/retries on ConnectionError when pinging workers
+    :return: dict with the following keys:
+        "services": list of status info for individual services
+        "software": list of softwares and their versions
+        "workers": list of status info for individual workers
+    :raises IIBError: if short is True and a problem is found
+    """
+    services = []
+
+    def add_status(service_name, ok, reason):
+        if short and not ok:
+            raise IIBError(f"{service_name} unavailable: {reason}")
+        service = {"name": service_name, "ok": ok}
+        if not ok:
+            service["reason"] = reason
+        services.append(service)
+
+    add_status("DATABASE", *database_connection_check())
+    rabbit_ok, rabbit_reason = rabbitmq_connection_check(get_worker_config().broker_url)
+    add_status("RABBITMQ", rabbit_ok, rabbit_reason)
+    add_status("QUAY", *quay_status_check())
+
+    if rabbit_ok:
+        workers = workers_status(retries=worker_ping_retries)
+    else:
+        workers = []
+
+    any_worker_ok = any(worker["ok"] for worker in workers)
```

Comment:
@yashvardhannanavati could you help me remember the conclusion on this topic. Thanks!

Reply:
Thinking about it now, we should probably fail even if one of the workers is no working. Here's why, CVP might have 6 workers and 1 of them might be down. But CVP doesn't know which of its request will be routed to that 1 unstable worker giving them a bad experience. We as a team should be proactive to react to it rather than ignoring it as a warning. 

So I think simply listing the status of the workers as @zanssa mentioned is sufficient, let's just make the health check fail on that. @zanssa I apologize for this last minute change but it just occured to me. I would be ok, if you want to cover this change in a separate ticket and keep this PR working as we discussed previously

Reply:
@yashvardhannanavati Thanks for this idea, it is easier to implement then the solution I had in mind.